File: js_code/167_two sum2.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(numbers, target) {
      let i=0,j=numbers.length-1,sum=0;
      let ans=[];
      while (i<j){
        sum=numbers[i]+numbers[j];
        if(sum==target) break;
        else if(sum<target) i++;
        else if(sum>target) j--;
      }
      ans.push(i+1);
      ans.push(j+1);
      console.log(ans);
      return ans;
};

twoSum([2,7,11,15],9)

File: py_code/167_two sum2.py
from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
      i,j,sum=0,len(numbers)-1,0
      ans=[]
      while i<j:
        sum=numbers[i]+numbers[j]
        if sum==target: break
        elif sum<target: i+=1
        elif sum>target: j-=1
      ans.append(i+1)
      ans.append(j+1)
      print(i,j);
      return ans
      pass

s=Solution()
s.twoSum([2,7,11,15],9)